### PR TITLE
Core roadmap June 2026

### PR DESCRIPTION
**Pull request by graciegoheen**
Roadmap post for June 2026 - Announcing… v2.0

### CODE REVIEW

File: docs/roadmap/2026-06-announcing-v2.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+# Announcing… v2.0 (June 2026)
+
+*At last.*
+
+Okay, so you saw the [announcement](https://www.getdbt.com/blog/fivetran-and-dbt-are-one-company-now-here-s-what-that-means), and you read [the dev blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here), which explain in detail how we got from there (Fusion, new engine, ELv2) to here (it’s just dbt Core, v2.0).
```
Comment (jtcohen6) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
Okay, so you saw the [announcement](https://www.getdbt.com/blog/fivetran-and-dbt-are-one-company-now-here-s-what-that-means), and you read [the dev blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here), which explains in detail how we got from there (Fusion, new engine, ELv2) to here (it’s just dbt Core, v2.0).
```

File: docs/roadmap/2026-06-announcing-v2.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+If you want to read more about how our thinking and licensing strategy evolved over the past year, [Joel & Grace wrote a whole blog post](https://docs.getdbt.com/blog/dbt-core-v2-is-here).
+
+**Why alpha?** Isn’t Fusion already in Preview? 
+
+We’ve spent the last 6+ months working hard on Fusion’s “conformance”: given the same project inputs, does it produce the same results as Core v1.X? We fixed thousands of bugs using this method, and while we now feel confident running Fusion in an environment we manage (dbt platform), software loves to find new edge cases out in the big bold world. 
+
+That's why we need YOUR help to run and test v2.0 against the full population of dbt projects. We need YOUR ever-bigger DAGs, YOUR wacky package macros, YOUR bespoke override of the incremental materialization, and YOUR 2009 laptop running Windows Vista.
+
+The best way to help is by [testing out v2.0](https://docs.getdbt.com/docs/local/install-dbt) while it’s in alpha - try running it against your project, and let us know if you encounter any unexpected errors
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
The best way to help is by [testing out v2.0](https://docs.getdbt.com/docs/local/install-dbt) while it’s in alpha - try running it against your project, and let us know if you encounter any unexpected errors.
```

File: docs/roadmap/2026-06-announcing-v2.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+In the meantime, we actually did the two-engine thing. We triaged bugs and feature requests. We refined them, shipped them, and tested them, *twice,* in both dbt Core v1.12 (Python) and Fusion / v2.0 (Rust). So, regardless of whether you’re still using Core or Fusion today, we’ve got good news for you - the dbt framework is alive and well.
+
+| **Version** | **When** | **Namesake** | **Stuff** |
+| --- | --- | --- | --- |
+| [v1.12](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.10) | May (beta) | Alecia Beth Moore-Hart (P!NK) | UDF extensions. Easier specs for semantic layer, and Iceberg `catalogs`. Long-upvoted feature requests: `vars.yml`; `on_error` model config; [mix-and-match yaml selectors](https://docs.getdbt.com/reference/node-selection/methods?version=1.12#selector); `run-operation` for adhoc `—-sql`; jinja file extensions; `latest_version_pointer`. Improved exception handling and error messages. Dozens of bug fixes across core + adapters, including Python 3.14 support. `dbt login` to unlock paid platform features. `--use-v2-parser` to try out the speedy new Rust parser before you upgrade to v2.0. |
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
| [v1.12](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.12) | May (beta) | Alecia Beth Moore-Hart (P!NK) | UDF extensions. Easier specs for semantic layer, and Iceberg `catalogs`. Long-upvoted feature requests: `vars.yml`; `on_error` model config; [mix-and-match yaml selectors](https://docs.getdbt.com/reference/node-selection/methods?version=1.12#selector); `run-operation` for adhoc `—-sql`; jinja file extensions; `latest_version_pointer`. Improved exception handling and error messages. Dozens of bug fixes across core + adapters, including Python 3.14 support. `dbt login` to unlock paid platform features. `--use-v2-parser` to try out the speedy new Rust parser before you upgrade to v2.0. |
```

File: docs/roadmap/2026-06-announcing-v2.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+| [v2.0](https://docs.getdbt.com/docs/dbt-versions/core-upgrade/upgrading-to-v1.11) | June (alpha) |  | Faster Rust rewrite, at full parity with v1. Strict codified language spec. New parquet metadata artifacts, built for speed & scale, powering new dbt-docs. |
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
| v2.0 | June (alpha) |  | Faster Rust rewrite, at full parity with v1. Strict codified language spec. New parquet metadata artifacts, built for speed & scale, powering new dbt-docs. |
```

File: docs/roadmap/2026-06-announcing-v2.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+## Present
+
+Today, we [announced the first alpha release of dbt Core v2.0](https://docs.getdbt.com/blog/dbt-core-v2-is-here). This new major version is powered by the same Rust rewrite of Core v1 that powers Fusion. All of the code is licensed under Apache 2.0, and you can see it on the `main` branch of the `dbt-core` repo. 
+
+This is the future of the dbt framework:
+
+- fast and built for scale (Rust, ADBC, etc)
+- shared foundation - one donut, not two donuts
+- single adapter layer - all bundled in
+- and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out the github discussion (link coming soon) for more details
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
- and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out [the github discussion](https://github.com/dbt-labs/dbt-core/discussions/13080) for more details
```

File: docs/roadmap/2026-06-announcing-v2.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+2. **Checks:** In the dbt-docs discussion (link coming soon), we describe how we’re powering fast local metadata queries with a new set of well-structured parquet files and an in-process database. Those same files could power *project quality checks* as metadata queries, folded into the task graph, executed during compile / before execution. You could prevent a `dbt build` if a model is missing an owner, messing up a naming convention, or selecting from a source it shouldn’t be. (For any SDF knowers out there, [this should sound familiar](https://docs.sdf.com/guide/data-quality/checks).) Imagine: a [v2.0 of the dbt-project-evaluator package](https://github.com/dbt-labs/dbt-project-evaluator) that ships our best practices as `checks`, written as simple SQL queries (instead of Jinja `{{ graph }}` manipulation that’s as impressive as it is illegible).
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
2. **Checks:** In [the dbt-docs discussion](https://github.com/dbt-labs/dbt-core/discussions/13080), we describe how we’re powering fast local metadata queries with a new set of well-structured parquet files and an in-process database. Those same files could power *project quality checks* as metadata queries, folded into the task graph, executed during compile / before execution. You could prevent a `dbt build` if a model is missing an owner, messing up a naming convention, or selecting from a source it shouldn’t be. (For any SDF knowers out there, [this should sound familiar](https://docs.sdf.com/guide/data-quality/checks).) Imagine: a [v2.0 of the dbt-project-evaluator package](https://github.com/dbt-labs/dbt-project-evaluator) that ships our best practices as `checks`, written as simple SQL queries (instead of Jinja `{{ graph }}` manipulation that’s as impressive as it is illegible).
```

File: docs/roadmap/2026-06-announcing-v2.md (hunk 5 above)
Comment (jtcohen6) [reply docs/roadmap/2026-06-announcing-v2.md]: https://github.com/dbt-labs/dbt-core/discussions/13080

Comment (jtcohen6) [inline docs/roadmap/2026-06-announcing-v2.md]:
```suggestion
- and one more thing… a brand-new dbt-docs (!) This is a long-awaited refresh (something we talked about in our [very first roadmap post](https://github.com/dbt-labs/dbt-core/blob/main/docs/roadmap/2022-05-dbt-a-core-story.md#epilogue-whats-missing)) - check out [the github discussion](https://github.com/dbt-labs/dbt-core/discussions/13080) for more details
```

**Review by graciegoheen — COMMENTED** [review]
Adding github discussion link

**Review by jtcohen6 — APPROVED** [review]
_If we shadows have offended, think but this and all is mended_

**Review by eliasdefaria — APPROVED** [review]
_We are such stuff as dreams are made on, and our little life is rounded with a sleep_

File: docs/roadmap/2026-06-announcing-v2.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+Our request for all of you:
+
+- be excited with us and talk about it
+- try this all out and tell us how to make it better
+- as we consolidate repos, help us make sure no long-loved issue is forgotten
+- shape the future of dbt by joining [discussions on GitHub](https://github.com/dbt-labs/dbt-core/discussions) 
+- and, hey, [come hangout in Vegas this September](https://www.getdbt.com/dbt-summit/registration)
+
+HAGS,
```
Comment (graciegoheen) [inline docs/roadmap/2026-06-announcing-v2.md:141]:
```suggestion
HAGS (have a great summer),
```

**Review by graciegoheen — COMMENTED** [review]
: )
